cmd/create/cluster/cmd.go

```diff
@@ -2035,21 +2043,46 @@ func run(cmd *cobra.Command, _ []string) {
 	private := args.private
 	isPrivateHostedCP := isHostedCP && private // all private hosted clusters are private-link
```

Comment:
`isPrivateHostedCP` can be initialized to false here even if the user sets private API to true later, I think we need to move this declaration further down or just set the variable to true if the user selects private API in the prompt